packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTableSet.tsx

```diff
@@ -0,0 +1,286 @@
+        />
+      );
+    }
+    return undefined;
+  }
+
+  public renderEach(): JSX.Element | undefined {
+    if (this.state.properties.length > 1) {
+      return (
+        <Stack gap={MEDIUM_GAP_SIZE}>
+          <ActionButton
+            iconProps={{ iconName: this.state.showSeeMore ? 'SkypeCircleMinus' : 'CirclePlus' }}
+            onClick={this._onClickSeeMore}
+            onRenderText={this._onRenderText}
+            styles={{ root: { paddingLeft: '0px' }, textContainer: { paddingLeft: '4px' } }}
+          >
+            See More
+          </ActionButton>
+          {this.state.showSeeMore && (
+            <Stack gap={LARGE_GAP_SIZE}>
+              {this.state.properties.map((item: IApiProperty, index: number) =>
+                index !== 0 ? (
+                  <ApiReferencesTable
+                    key={item.propertyName}
+                    name={item.propertyName}
+                    title={item.title}
+                    description={item.description}
+                    extendsTokens={item.extendsTokens}
+                    properties={item.property}
+                    methods={item.methods}
+                    renderAsEnum={item.propertyType === PropertyType.enum}
+                    renderAsClass={item.propertyType === PropertyType.class}
+                  />
+                ) : (
+                  undefined
+                )
+              )}
+            </Stack>
+          )}
+        </Stack>
+      );
+    }
+    return undefined;
+  }
+
+  public render(): JSX.Element {
+    const { className } = this.props;
+    return (
+      <Stack gap={LARGE_GAP_SIZE} className={css(apiReferencesTableTopMargin, className)}>
+        {this.renderFirst()}
+        {this.renderEach()}
+      </Stack>
+    );
+  }
+
+  public componentDidMount(): void {
+    window.addEventListener('hashchange', this._onHashChange);
+  }
+
+  public componentWillUnmount(): void {
+    window.removeEventListener('hashchange', this._onHashChange);
+  }
+
+  public componentDidUpdate(prevProps: IApiReferencesTableSetProps, prevState: IApiReferencesTableSetState): void {
+    if (prevState.showSeeMore === false && this.state.showSeeMore === true) {
+      const hash = this._extractAnchorLink(window.location.hash);
+      const el = document.getElementById(hash);
+
+      if (el) {
+        // update scroll position
+        window.scrollTo({ top: el.offsetTop - 31.5 /* title */, behavior: 'smooth' });
+      }
+    }
+  }
+
+  private _onHashChange = (): void => {
+    const { properties, showSeeMore } = this.state;
+
+    const hash = this._extractAnchorLink(window.location.hash);
+    const el = document.getElementById(hash);
+
+    if (el) {
+      // update scroll position
+      window.scrollTo({ top: el.offsetTop - 31.5 /* title */, behavior: 'smooth' });
+    }
+
+    if (!showSeeMore) {
+      const section = properties.filter(x => x.propertyName === hash)[0];
```

Comment:
Are we guaranteed a non-empty array here?

Reply:
Filter will work even if the array is empty (it will just return another empty array). Then accessing `[0]` will return undefined. (in JS, accessing an array index that doesn't exist is just like accessing any other object property that doesn't exist: it returns undefined and doesn't throw)

Reply:
I was concerned about `section` potentially being undefined, but I see that it is guarded against below.